framework/src/araui/core/ui_core.c

```diff
@@ -396,15 +399,21 @@ static void _ui_update_redraw_list_recur(ui_widget_body_t *widget)
 			}
 
 			// Update previous area
-			ui_window_add_redraw_list(curr_widget->global_rect);
+			if (ui_window_add_redraw_list(curr_widget->global_rect) != UI_OK) {
```

Comment:
don't have to check result of translate & rotate & scale?

Reply:
Yes, they just do multiplication of float number.